src/br/univali/portugol/nucleo/bibliotecas/graficos/SuperficieDesenhoImpl.java

```diff
@@ -30,7 +30,7 @@
 {
     private final PoolOperacoesGraficas POOL_OPERACOES_GRAFICAS = new PoolOperacoesGraficas();
 
-    private final OperacaoGrafica[] operacoes = new OperacaoGrafica[2048];
+    private final OperacaoGrafica[] operacoes = new OperacaoGrafica[131072];

```

Comment:
@noschang , se usar uma lista é importante lembrar de não usar iterador (implícita (com for each) ou explicitamente) para evitar zilhões de instâncias de iteradores alocadas durante a execução.